src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/Marshallers/ComInterfaceDispatchMarshallerFactory.cs

```diff
@@ -0,0 +1,68 @@
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Interop
+{
+    internal sealed record ComInterfaceDispatchMarshallingInfo : MarshallingInfo
+    {
+        public static readonly ComInterfaceDispatchMarshallingInfo Instance = new();
+    }
+
+    internal sealed class ComInterfaceDispatchMarshallerFactory : IMarshallingGeneratorFactory
+    {
+        private readonly IMarshallingGeneratorFactory _inner;
+        public ComInterfaceDispatchMarshallerFactory(IMarshallingGeneratorFactory inner)
+        {
+            _inner = inner;
+        }
+
+        public IMarshallingGenerator Create(TypePositionInfo info, StubCodeContext context)
+            => info.MarshallingAttributeInfo is ComInterfaceDispatchMarshallingInfo ? new Marshaller() : _inner.Create(info, context);
+
+        private sealed class Marshaller : IMarshallingGenerator
+        {
+            public ManagedTypeInfo AsNativeType(TypePositionInfo info) =>
+                new PointerTypeInfo(
+                    $"{TypeNames.System_Runtime_InteropServices_ComWrappers_ComInterfaceDispatch}*",
+                    $"{TypeNames.System_Runtime_InteropServices_ComWrappers_ComInterfaceDispatch}*",
+                    IsFunctionPointer: false);
+            public IEnumerable<StatementSyntax> Generate(TypePositionInfo info, StubCodeContext context)
+            {
+                if (context.CurrentStage != StubCodeContext.Stage.Unmarshal)
+                {
+                    yield break;
+                }
+
+                var (managed, native) = context.GetIdentifiers(info);
+
+                // <managed> = ComWrappers.ComInterfaceDispatch.GetInstance<<managedType>>(<native>);
+                yield return ExpressionStatement(
+                    AssignmentExpression(SyntaxKind.SimpleAssignmentExpression,
+                        IdentifierName(managed),
+                        InvocationExpression(
+                            MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                                ParseName(TypeNames.System_Runtime_InteropServices_ComWrappers_ComInterfaceDispatch),
+                                GenericName(
+                                    Identifier("GetInstance"),
+                                    TypeArgumentList(SingletonSeparatedList(info.ManagedType.Syntax)))),
+                            ArgumentList(
+                                SingletonSeparatedList(
+                                    Argument(
+                                        IdentifierName(native)))))));
+            }
+
+            public SignatureBehavior GetNativeSignatureBehavior(TypePositionInfo info) => SignatureBehavior.NativeType;
+            public ValueBoundaryBehavior GetValueBoundaryBehavior(TypePositionInfo info, StubCodeContext context) => ValueBoundaryBehavior.NativeIdentifier;
+            public bool IsSupported(TargetFramework target, Version version) => true;
```

Comment:
Seems like this needs to be limited to .NET 8, right?

Reply:
We can limit this to .NET 5 or newer as that's the first version where the API we're using here was introduced.